frontend/pages/SoftwarePage/SoftwareVulnerabilities/SoftwareVulnerabilities.tsx

```diff
@@ -66,23 +89,167 @@ const SoftwareVulnerabilities = ({
     () => getVulnerabilities(queryParams),
     {
       keepPreviousData: true,
-      staleTime: 30000,
+      enabled: !isExactMatchQuery && isSoftwareEnabled,
+      onSuccess: (data) => {
+        setTableData(data);
+        if (data.count === 0) {
+          if (
+            queryParams.exploit ||
+            (queryParams.query && queryParams.query.length > 0)
+          ) {
+            setEmptyStateReason("no-matching-items");
+          } else {
+            setEmptyStateReason("no-vulns-detected");
+          }
+        }
+      },
     }
   );
 
-  if (isLoading) {
+  // Calling software/vulnerabilities/:CVE endpoint when user searches with quotation marks
+  const {
+    isLoading: isLoadingExactMatch,
+    isFetching: isFetchingExactMatch,
+    refetch: refetchExactMatch,
+  } = useQuery<
+    IVulnerabilityResponse | null,
+    AxiosError<IApiError>,
+    IVulnerabilityResponse,
+    IGetVulnerabilityQueryKey[]
+  >(
+    [
+      {
+        scope: "softwareVulnByCVE",
+        vulnerability: (query && stripQuotes(query)) || "",
+        teamId,
+      },
+    ],
+    ({ queryKey }) => {
+      return softwareVulnAPI.getVulnerability(queryKey[0]);
+    },
+    {
+      ...DEFAULT_USE_QUERY_OPTIONS,
+      retry: false,
+      onSuccess: (data) => {
+        // Handle 204 response which doesn't return data if it is a known CVE but doesn't exist in response
+        if (!data) {
+          setTableData({
+            count: 0,
+            counts_updated_at: "",
+            vulnerabilities: [],
+            meta: {
+              has_next_results: false,
+              has_previous_results: false,
+            },
+          });
+          setEmptyStateReason("known-vuln");
+        }
+        // If filtering for exploited vulns, hide vulnerability if cisa_known_exploit is false
+        else if (
+          queryParams.exploit &&
+          !data.vulnerability.cisa_known_exploit
+        ) {
+          setTableData({
+            count: 0,
+            counts_updated_at: "",
+            vulnerabilities: [],
+            meta: {
+              has_next_results: false,
+              has_previous_results: false,
+            },
```

Comment:
Since this object is repeated so many times would it be worth storing it in a variable?

Reply:
Yeah, I set the object in `tableData`. Is that what you're suggesting?